src/main/java/net/dv8tion/jda/core/JDABuilder.java

```diff
@@ -472,8 +472,8 @@ public JDABuilder useSharding(int shardId, int shardTotal)
     {
         if (accountType != AccountType.BOT)
             throw new AccountTypeException(AccountType.BOT);
-        if (shardId < 0 || shardTotal < 1 || shardId >= shardTotal)
-            throw new RuntimeException("This configuration of shardId and shardTotal is not allowed! 0 <= shardId < shardTotal with shardTotal > 0");
+        Checks.check(shardId >= 0 && shardTotal > 0 && shardId < shardTotal,
```

Comment:
Boolean logic was changed, but the error message was not updated.
Also, the error message should be clarified. It is a bit complicated.

Reply:
The logic has actually not changed it was just inverted here